### PR TITLE
MINIFI-188: Incorporate CATCH and example tests

### PR DESCRIPTION
**Pull request by phrocker**
This PR introduces catch and a few example tests.
Additional tests will follow, but they currently cause
several segfaults without a large number of corresponding
changes in the code. Therefore to minimize the bootstrapping
I will include the cmake changes in addition to the dep.

to run type make; make test 

You can also run ./tests in the build directory for more information

CATCH is Boost license, let me know if this is a problem. Thanks!

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: libminifi/cmake/FindLeveldb.cmake [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless qrequired by applicable law or agreed to in writing,
```
Comment (apiri) [inline libminifi/cmake/FindLeveldb.cmake]: typo

Comment (phrocker) [inline libminifi/cmake/FindLeveldb.cmake]: Thanks! I'll fix that.

**Comment by apiri** [comment]
reviewing updates

File: libminifi/cmake/FindLeveldb.cmake (hunk 1 above)
Comment (phrocker) [inline libminifi/cmake/FindLeveldb.cmake]: I see this same typo in other files. I'll make that change in either this PR or the next. Apparently I propagated the typo.

Comment (apiri) [inline libminifi/cmake/FindLeveldb.cmake]: Ah, gotcha.  Good with this one or a follow-on.  Your call!

Comment (apiri) [inline libminifi/cmake/FindLeveldb.cmake]: Actually, I can just take care of those items as part of the merge.

Comment (phrocker) [inline libminifi/cmake/FindLeveldb.cmake]: Oh I already force pushed the change.

Comment (phrocker) [inline libminifi/cmake/FindLeveldb.cmake]: I updated the commit to include a blurb about it so that it was squashed into a single commit.

**Comment by apiri** [comment]
@phrocker Thanks for cleaning up the typos.  The changes all look good and builds are working on Linux for me.

Boost license is perfectly fine as per http://www.apache.org/legal/resolved.html#category-a

To that end, and apologize for overlooking this initially, but we will need to update the project LICENSE as we are including the source as part of the codebase.  After that, I believe we should be good for merging.

Thanks!

**Comment by apiri** [comment]
Thanks for taking care of licensing.  Will get this merged in!

**Pull request closed without merging**